src/js/actions/menu.js

```diff
@@ -570,6 +570,181 @@ define(function (require, exports) {
+                    return result;
+                });
+            };
+
+            descriptor.batchPlay = function (commands, options, batchOptions) {
+                var commandNames = commands.map(function (c) { return c.name; }).join(", "),
+                    isHitTest = commandNames === "hitTest";
+
+                if (!isHitTest) {
+                    var str = "(" + JSON.stringify(commands, null, " ") + ", " +
+                        JSON.stringify(options, null, " ") + ", " +
+                        JSON.stringify(batchOptions, null, " ") + ");";
+                    log.debug("[Descriptor] batchPlay: %c%s\n%c%s", eventStyle, commandNames, blockStyle, str);
+                }
+                return descriptor.__batchPlay(commands, options, batchOptions).then(function (result) {
+                    if (!isHitTest) {
+                        log.debug("[Descriptor] batchPlay result: \n%c%s", blockStyle,
+                            JSON.stringify(result, null, "  "));
+                    }
+                    return result;
+                });
+            };
+
+            descriptor.get = function (reference) {
+                log.debug("[Descriptor] get: \n%c%s", blockStyle, JSON.stringify(reference, null, ""));
+                return descriptor.__get(reference).then(function (result) {
+                    log.debug("[Descriptor] get result: \n%c%s", blockStyle, JSON.stringify(result, null, "  "));
+                    return result;
+                });
+            };
+            
+            descriptor.on("all", descriptor.__eventHandler);
+        } else {
+            descriptor.play = descriptor.__play;
+            descriptor.batchPlay = descriptor.__batchPlay;
+            descriptor.get = descriptor.__get;
+            descriptor.off("all", descriptor.__eventHandler);
+        }
+
+        return this.transfer(preferencesActions.setPreference, "descriptorLoggingEnabled", enabled);
+    };
+    logDescriptor.action = {
+        reads: [],
+        writes: [locks.JS_PREF],
+        transfers: [preferencesActions.setPreference]
+    };
+    
+    /**
+     * Debug-only method to toggle Headlights logging
+     * 
+     * @param {Object} options
+     * @param {Object} options.toggle
+     * @return {Promise}
+     */
+    var logHeadlights = function (options) {
+        if (!__PG_DEBUG__) {
+            return Promise.resolve();
+        }
+        
+        options = _.merge({ toggle: true }, options);
+
+        var preferencesStore = this.flux.store("preferences"),
+            preferences = preferencesStore.getState(),
+            enabled = preferences.get("headlightsLoggingEnabled"),
+            headlights = require("js/util/headlights");
+        
+        if (options.toggle) {
+            enabled = !enabled;
+        }
+        
+        headlights._logEvent = headlights._logEvent || headlights.logEvent;
```

Comment:
Above you stored these temporary properties with two underscores, but here you're only using one. They should be consistent.